src/authentication/team-token.ts

```diff
@@ -0,0 +1,88 @@
+import * as Knex from 'knex';
+import * as moment from 'moment';
+
+const randomstring = require('randomstring');
+
+export const teamTokenLength = 16;
+
+export interface TeamToken {
+  teamId: number;
+  token: string;
+  createdAt: number | moment.Moment;
+}
+
+/**
+ * Creates an sql query for fetching valid, i.e. latest, team tokens.
+ * The results can be filtered by specifying a token,
+ * a teamId or both.
+ */
+export function teamTokenQuery(db: Knex, token?: string, teamId?: number) {
+  const latestTokens = db('teamtoken')
+    .select('teamId')
+    .max('createdAt AS latestStamp')
+    .groupBy('teamId')
+    .as('latest');
+  const query = db('teamtoken')
+    .join(latestTokens, ((join: any) => join
+      .on('teamtoken.teamId', '=', 'latest.teamId')
+      .andOn('teamtoken.createdAt', '=', 'latest.latestStamp')) as any,
+    );
+  if (token) {
+    if (token.length !== teamTokenLength || !token.match(/^\w+$/)) {
```

Comment:
I think it's a good practice to have checks for the assumptions we're allowed to make about the input. If we really need to change the token length or format without breaking the existing ones, it's doable at that point by adding a version field to the tokens.